src/get-virtualization-type.js

```diff
@@ -0,0 +1,39 @@
+let log = require('./log');
+let debug = log.debugCompat('aws-provisioner:lib:check-for-ami');
+
+/**
+ * Check for the existence of an AMI that's executable by
+ * the provided ec2 object in the given region
+ */
+module.exports = async function (ec2, ami) {
+  let request = {
+    /* Not sure why this isn't working
+    ExecutableUsers: [
+      'self',
+      'all',
+    ],*/
+    ImageIds: [ami],
+  };
+
+  let result;
+  try {
+    result = await ec2.describeImages(request).promise();
+    debug('loaded information about %s', ami);
+  } catch (err) {
+    debug('did not find %s', ami);
+    return false;
+  }
+
+  if (result.data.Images.length === 0) {
+    throw new Error('Image does not exist');
+  } else if (result.data.Images.length > 1) {
+    debug(result.data);
+    throw new Error('Image returned more than one result');
+  }
+
+  if (result.data.Images[0].ImageId === ami) {
+    return result.data.Images[0].VirtualizationType;
```

Comment:
You could learn how to do a rebase!  (not because I'm lazy, but it's good experience).

Here's what I'd do, roughly.
```bash
git checkout -t <mygoodlocalbranch> -b rebase-on-changes-22-8-2016
git rebase origin/changes-22-8-2016
```
If it doesn't print any errors, do `git diff origin/changes-22-8-2016` to verify things worked as you expected.  If it does have errors you can fix them up with this: https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/